Engine/Component/ComponentParticleSystem.cpp

```diff
@@ -163,53 +193,83 @@ void ComponentParticleSystem::Render()
 			}
 		}
 
+		//update velocity over time
+		if (velocity_over_time && type_of_velocity_over_time == LINEAR)
+		{
+			acceleration = (velocity_particles_start * velocity_over_time_speed_modifier_second -
+				velocity_particles_start * velocity_over_time_speed_modifier) / particles_life_time;
+		}
+
+
+		if (gravity)
+			gravity_vector = float4 (0, gravity_modifier / 10000000, 0,0);
+
+		glUseProgram(shader_program);
+
 		// update all particles
+		int num_of_alive_particles = 0;
 		for (unsigned int i = 0; i < playing_particles_number; ++i)
 		{
 			Particle& p = particles[i];
-		
+
 			if (p.life > 0.0f)
 			{
 				UpdateParticle(p);
-				if (follow_owner)
-				{
-					billboard->Render(owner->transform.GetGlobalTranslation() + (p.rotation *p.position));
-				}
-				else
-				{
-					billboard->Render(p.position);
-				}
-				
+				std::iter_swap(particles.begin() +i, particles.begin()+ num_of_alive_particles);
+				++num_of_alive_particles;
 			}
+
 		}
+		billboard->CommonUniforms(shader_program);
```

Comment:
the uniforms should be on this cpp

Reply:
Dont worry about this too much, as this changes with uber